src/app/OperationalSessionSetup.h

```diff
@@ -155,6 +155,31 @@ typedef void (*OnDeviceConnectionRetry)(void * context, const ScopedNodeId & pee
 class DLL_EXPORT OperationalSessionSetup : public SessionEstablishmentDelegate, public AddressResolve::NodeListener
```

Comment:
So as this extends `SessionEstablishmentDelegate`, can we not extend that delegate to have a method (with a default-implemented {}) which is something like `connectionStateUpdated`, which passes the connection state, called on every state change?